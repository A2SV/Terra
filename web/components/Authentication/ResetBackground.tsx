import React from "react";
import backgroundImage from "../../public/background.png";
import Image from "next/image";
import logo from "../../public/logo.png";

const ResetBackground: React.FC = () => {
  return (
    <div className="flex items-center justify-center min-h-screen bg-white">
      <div className="flex md:flex-row bg-white shadow-lg min-h-screen w-full md:w-auto">
        <div className="hidden md:block relative w-1/2 overflow-hidden rounded-2xl p-6">
          <Image
            src={backgroundImage}
            alt="Background"
            className="object-cover rounded-2xl h-full w-full  ml-2"
          />
          <div className="absolute bottom-10 left-20 right-0 p-8 font-poppins">
            <h2 className="text-2xl font-semibold text-white">Forgot Password?</h2>
            <p className="mt-2 text-white">Reset password</p>
          </div>
        </div>
        <div className="w-full items-center md:w-1/2">
          <div className="flex justify-center flex-row md:justify-center font-nunito">
            <Image src={logo} alt="Logo" height={180} width={150} className="object-cover" />
          </div>
        </div>
      </div>
    </div>
  );
};

export default ResetBackground;
