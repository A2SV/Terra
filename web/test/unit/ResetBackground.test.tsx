import React from "react";
import { render, screen } from "@testing-library/react";
import ResetBackground from "@/components/Authentication/ResetBackground";

describe("ResetBackground", () => {
  it("renders without crashing", () => {
    render(<ResetBackground />);
    expect(screen.getByText(/Forgot Password\?/i)).toBeInTheDocument();
    expect(screen.getByText(/Reset password/i)).toBeInTheDocument();
  });

  it("renders exactly two images", () => {
    render(<ResetBackground />);
    const images = screen.getAllByRole("img");
    expect(images).toHaveLength(2);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ResetBackground />);
    expect(asFragment()).toMatchSnapshot();
  });
});
