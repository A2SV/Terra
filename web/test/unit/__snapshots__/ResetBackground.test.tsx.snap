// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetBackground matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center min-h-screen bg-white"
  >
    <div
      class="flex md:flex-row bg-white shadow-lg min-h-screen w-full md:w-auto"
    >
      <div
        class="hidden md:block relative w-1/2 overflow-hidden rounded-2xl p-8"
      >
        <img
          alt="Background"
          class="object-cover rounded-2xl h-full w-full  ml-2"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div
          class="absolute bottom-10 left-20 right-0 p-8 font-poppins"
        >
          <h2
            class="text-2xl font-semibold text-white"
          >
            Forgot Password?
          </h2>
          <p
            class="mt-2 text-white"
          >
            Reset password
          </p>
        </div>
      </div>
      <div
        class="w-full items-center md:w-1/2"
      >
        <div
          class="flex justify-center flex-row md:justify-center font-nunito"
        >
          <img
            alt="Logo"
            class="object-cover"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
